scalanet/src/io/iohk/scalanet/peergroup/dynamictls/ChannelAwareQueue.scala

```diff
@@ -0,0 +1,63 @@
+package io.iohk.scalanet.peergroup.dynamictls
+
+import io.iohk.scalanet.peergroup.CloseableQueue
+import io.netty.channel.ChannelConfig
+import monix.eval.Task
+import monix.execution.ChannelType
+
+import java.util.concurrent.atomic.AtomicLong
+
+/**
+  * Wraps an underlying unbounded CloseableQueue queue and bounds it based on netty auto-read feature.
+  * While auto-read is disabled received messages start to accumulate in underlying os RCV tcp buffer. When RCV buffer is full
+  * sender SND buffer will start to buffer un-sent bytes. When sender SND buffer is full, the default behaviour is that
+  * write(xxx) will block.  In our case, sendMessage Task will not finish until there will be place in SND buffer
+  *
+  *
+  * WARNING: Actual limit may sometimes goes higher, as each netty read can return more than one message.
+  *
+  * @param limit how many items can accumulate in the queue
+  * @param queue is the underlying closeable message queue
+  */
+private[scalanet] final class ChannelAwareQueue[M] private (limit: Int, queue: CloseableQueue[M]) {
+  private val queueSize = new AtomicLong(0)
+
+  private val lowerBound: Int = Math.max(1, limit / 2)
+
+  def size: Long = queueSize.get()
+
+  def offer(channelConfig: ChannelConfig, a: M): Task[Unit] = {
+    Task(enableBPIfNecessary(channelConfig)).flatMap(_ => queue.offer(a).void)
```

Comment:
```suggestion
    Task(enableBPIfNecessary(channelConfig)) >> queue.offer(a)
```

Reply:
tbh In general i am not great fan of additional symbols (I always forget what they mean and need to go over the docs) but lets say this one is ubiquitous enough
My red flag is when I need to import whole `cats.implicits._` scope to to use some symbol.

Reply:
You can be more specific by importing from `cats.syntax` or `cats.instances` but that requires more familiarity, see [here](https://typelevel.org/cats/typeclasses/imports.html) and [here](https://typelevel.org/cats/jump_start_guide.html).

Reply:
you can, but imo it is bad trade off with most of the symbols, as you force this familiarity onto future maintainers (and upon yourself) in most cases just to save few keystrokes.

Reply:
That's why I don't do it, and also because of the implicit resolution conflicts. `cats.implicits._` gives you everything (like `.void` for that matter).

It's not the key strokes I think are worth saving here, it just looks and reads much nicer. I don't know esoteric ones like `><>` whatever but some basics like `*>` and `<*` are great to clean up the code a bit.